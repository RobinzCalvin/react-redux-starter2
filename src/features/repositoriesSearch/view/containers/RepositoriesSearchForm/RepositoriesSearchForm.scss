.repositories-search-form {
  &__button {
    margin-top: 0.8rem;
    text-align: center;
  }
}
