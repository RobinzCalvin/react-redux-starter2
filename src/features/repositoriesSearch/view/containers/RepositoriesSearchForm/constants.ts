import { makeFormFieldNames } from 'shared/helpers';
import { IRepositoriesSearchFormFields } from '../../../namespace';

export const fieldNames = makeFormFieldNames<IRepositoriesSearchFormFields>([
  'searchString', 'forksNumber', 'owner', 'starsNumber', 'language',
]);
