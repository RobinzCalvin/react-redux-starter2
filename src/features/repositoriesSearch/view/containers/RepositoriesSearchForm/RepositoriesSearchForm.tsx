import * as React from 'react';
import { bindActionCreators, Dispatch } from 'redux';
import { connect } from 'react-redux';

import { IAppReduxState } from 'shared/types/app';
import { SearchForm } from 'shared/view/components';
import RepositoriesSearchSettings from './RepositoriesSearchSettings/RepositoriesSearchSettings';

import { selectors, actions } from './../../../redux';
import { IRepositoriesSearchFormFields } from '../../../namespace';
import { fieldNames } from './constants';

interface IOwnProps {
  onSubmit(formValues: IRepositoriesSearchFormFields): void;
}

interface IStateProps {
  isRepositoriesSearchRequesting: boolean;
}

interface IActionProps {
  searchRepositories: typeof actions.searchRepositories;
  resetSearchResults: typeof actions.resetSearchResults;
}

type IProps = IOwnProps & IStateProps & IActionProps;

function mapDispatch(dispatch: Dispatch): IActionProps {
  return bindActionCreators({
    searchRepositories: actions.searchRepositories,
    resetSearchResults: actions.resetSearchResults,
  }, dispatch);
}

function mapState(state: IAppReduxState): IStateProps {
  return {
    isRepositoriesSearchRequesting: selectors.selectCommunication(state, 'searchRepositories').isRequesting,
  };
}

function RepositoriesSearchForm(props: IProps) {
  const { isRepositoriesSearchRequesting, resetSearchResults } = props;
  return (
    <SearchForm<IRepositoriesSearchFormFields>
      searchInputName={fieldNames.searchString}
      isSearchRequesting={isRepositoriesSearchRequesting}
      onSubmit={handleFormSubmit}
      resetSearchResults={resetSearchResults}
      renderSettings={RepositoriesSearchSettings}
    />
  );

  function handleFormSubmit(formValues: IRepositoriesSearchFormFields) {
    const { searchRepositories, onSubmit } = props;
    searchRepositories({ searchOptions: formValues, page: 1 });
    onSubmit(formValues);
  }
}

export default connect(mapState, mapDispatch)(RepositoriesSearchForm);
