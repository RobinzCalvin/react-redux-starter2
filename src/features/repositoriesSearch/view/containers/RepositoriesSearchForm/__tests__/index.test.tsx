import React from 'react';
import { render } from 'enzyme';

import { RepositoriesSearchForm, IRepositoriesSearchFormProps } from '../RepositoriesSearchForm';

const props: IRepositoriesSearchFormProps = {
  onSubmit: jest.fn(),
  searchRepositories: jest.fn(),
  resetSearchResults: jest.fn(),
  isRepositoriesSearchRequesting: false,
};

describe('RepositoriesSearchForm component', () => {
  const component = render(<RepositoriesSearchForm {...props} />);
  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
