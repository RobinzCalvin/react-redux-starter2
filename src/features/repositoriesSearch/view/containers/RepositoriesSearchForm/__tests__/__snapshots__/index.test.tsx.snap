// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoriesSearchForm component should match snapshot 1`] = `
<form
  class="search-form"
>
  <div
    class="MuiFormControl-root-1 MuiFormControl-fullWidth-4"
  >
    <div
      class="MuiInputBase-root-18 MuiInput-root-5 MuiInput-underline-9 MuiInputBase-fullWidth-27 MuiInput-fullWidth-12 MuiInputBase-formControl-19 MuiInput-formControl-6"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input-28 MuiInput-input-13"
        name="searchString"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    class="search-form__buttons"
  >
    <button
      class="MuiButtonBase-root-61 MuiButton-root-35 MuiButton-outlined-43"
      tabindex="0"
      type="submit"
    >
      <span
        class="MuiButton-label-36"
      >
        Search
      </span>
    </button>
    <div
      class="search-form__settings"
    >
      <button
        class="MuiButtonBase-root-61 MuiButton-root-35 MuiButton-outlined-43"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-36"
        >
          Settings
        </span>
      </button>
    </div>
  </div>
</form>
`;
